src/app/public/modules/timepicker/timepicker.directive.ts

```diff
@@ -170,7 +170,9 @@ export class SkyTimepickerInputDirective implements
   public writeValue(value: any) {
     this.modelValue = this.formatter(value);
     this._validatorChange();
-    this._onChange(this.modelValue);
+    if (value !== this.modelValue) {
```

Comment:
I think we will need to do this in a way that the check happens before line 171. You will probably need to set the `this.formatter(value)` to a variable and then check that against `this.modelValue` before assigning it to `this.modelValue`

Reply:
But the _onChange is for the benefit of `FormControl`, it should be fired when the value changes from what `FormControl` thinks it is. So when it fires the writeValue with its value and we change it, that's when I call _onChange. I shouldn't need to trigger onChange when our internal value changes from a writeValue call that originated with `FormControl`